src/sagemaker/local/image.py

```diff
@@ -525,3 +545,24 @@ def _aws_credentials(session):
 def _write_json_file(filename, content):
     with open(filename, 'w') as f:
         json.dump(content, f)
+
+
+def _ecr_login_if_needed(boto_session, image):
+    # Only ECR images need login
+    if not ('dkr.ecr' in image and 'amazonaws.com' in image):
+        return
+
+    # do we have the image?
+    if _check_output('docker images -q %s' % image).strip():
+        return
+
+    ecr = boto_session.client('ecr')
+    auth = ecr.get_authorization_token(registryIds=[image.split('.')[0]])
+    authorization_data = auth['authorizationData'][0]
+
+    raw_token = base64.b64decode(authorization_data['authorizationToken'])
+    token = raw_token.decode('utf-8').strip('AWS:')
+    ecr_url = auth['authorizationData'][0]['proxyEndpoint']
+
+    cmd = "docker login -u AWS -p %s %s" % (token, ecr_url)
+    subprocess.check_output(cmd, shell=True)
```

Comment:
Sometimes we are using **_check_output** and sometimes we are using **subprocess.checkoutput**. Can we just use one of them?